contracts/src/IBoundlessMarketCallback.sol

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2025 RISC Zero, Inc.
+//
+// All rights reserved.
+pragma solidity ^0.8.24;
+
+/// @title IBoundlessMarketCallback
+/// @notice Interface for handling proof callbacks from BoundlessMarket with proof verification
+/// @dev Inherit from this contract to implement custom proof handling logic for BoundlessMarket proofs
+interface IBoundlessMarketCallback {
+    /// @notice Handles submitting proofs with RISC Zero proof verification
+    /// @dev If not called by BoundlessMarket, the function MUST verify the proof before proceeding.
```

Comment:
Even if the BoundlessMarket is the source, we should be strongly encouraging receivers to verify the seal. If the BoundlessMarket is compromised, we want to avoid this potentially leading to a soundness/safety failure in consuming applications (e.g. it should not be possible to drain the funds of a validity rollup that uses Boundless).

This does cost extra gas, however it is not a huge amount since the proof will usually be a Merkle inclusion path.

We will also want to be extra careful from now on about what external calls the BoundlessMarket contract makes.

Reply:
I'm fine with us recommending an ultra-safe approach in our sample callback, but not sure I follow the reasoning here, lets chat offline